### PR TITLE
Use the reinit function for MaterialModelInputs

### PR DESCRIPTION
**Pull request by anne-glerum**
This PR uses the reinit function to reinitialize the MaterialModel inputs. It shouldn't change anything, but it's late. 

### For all pull requests:

* [x] I have followed the [instructions for indenting my code](../blob/master/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [ ] I have tested my new feature locally to ensure it is correct.
* [ ] I have [created a testcase](http://www.math.clemson.edu/~heister/manual.pdf#sec%3Awriting_tests) for the new feature/benchmark in the [tests/](../blob/master/tests/) directory.
* [ ] I have added a changelog entry in the [doc/modules/changes](../blob/master/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

File: source/boundary_temperature/dynamic_core.cc [hunk 1]
```diff
@@ -693,30 +693,12 @@ namespace aspect
                 if (cell->face(f)->boundary_id() == CMB_id)
                   {
                     fe_face_values.reinit (cell, f);
+
+                    // Repopulate MaterialInputs, set use_strain_rates to false
+                    in.reinit(fe_face_values, cell, this->introspection(), this->get_solution(), false);
```
Comment (tjhei) [inline source/boundary_temperature/dynamic_core.cc:698]: I prefer writing this as
```suggestion
                    in.reinit(fe_face_values, cell, this->introspection(), this->get_solution(), /* use_straim_rates = */ false);
```
Then you can delete the comment above.

**Review by tjhei — APPROVED** [review]

File: source/boundary_temperature/dynamic_core.cc (hunk 1 above)
Comment (bangerth) [reply source/boundary_temperature/dynamic_core.cc:698]: I'm renaming this argument to `compute_strain_rates` in #4267. Might want to use that naming scheme.

File: source/boundary_temperature/dynamic_core.cc [hunk 2]
```diff
@@ -693,30 +693,12 @@ namespace aspect
...
+
                     fe_face_values[this->introspection().extractors.temperature].get_function_gradients (this->get_solution(),
                         temperature_gradients);
-                    fe_face_values[this->introspection().extractors.temperature].get_function_values (this->get_solution(),
-                        in.temperature);
```
Comment (bangerth) [inline source/boundary_temperature/dynamic_core.cc:699]: I guess all of these fields are unused? Or does the code in `reinit()` already set these variables correctly?

**Comment by bangerth** [comment]
@anne-glerum Any news?

File: source/boundary_temperature/dynamic_core.cc (hunk 2 above)
Comment (anne-glerum) [reply source/boundary_temperature/dynamic_core.cc:699]: `reinit` indeed sets these variables correctly, that's why I'm making this change.

**Review by bangerth — APPROVED** [review]

**Pull request merged**